test/units/TEST-71-HOSTNAME.sh

```diff
@@ -138,6 +142,44 @@ EOF
     assert_not_in 'Firmware Date' "$(hostnamectl)"
 }
 
+testcase_hardware_serial() {
+    # No DMI on s390x or ppc
+    if [[ ! -d /sys/class/dmi/id ]]; then
+        echo "/sys/class/dmi/id not found, skipping firmware date tests."
+        return 0
+    fi
+
+    trap restore_sysfs_dmi RETURN
+
+    fake_sysfs_dmi
+
+    echo '1234' >/sys/class/dmi/id/board_serial
+    stop_hostnamed
+    assert_eq "$(hostnamectl --json=short | jq --raw-output .HardwareSerial)" "1234"
+
+    # product_serial is preferred over board_serial
+    echo '4321' >/sys/class/dmi/id/product_serial
+    stop_hostnamed
+    assert_eq "$(hostnamectl --json=short | jq --raw-output .HardwareSerial)" "4321"
+}
+
+testcase_chassis_asset_tag() {
+    # No DMI on s390x or ppc
+    if [[ ! -d /sys/class/dmi/id ]]; then
+        echo "/sys/class/dmi/id not found, skipping firmware date tests."
+        return 0
+    fi
+
+    trap restore_sysfs_dmi RETURN
+
+    fake_sysfs_dmi
+
+    echo 'secret' >/sys/class/dmi/id/chassis_asset_tag
```

Comment:
This will not work, as udevd will not read the value.

For other test cases, e.g. testcase_hardware_serial, the sysattr is directly read by hostname, hence they work. But, in this case, the sysattr must be read by udevd, and hostnamed reads the sysattr through the udev property set by udevd.